week3/createContentTrainingData.py

```diff
@@ -52,7 +59,8 @@ def transform_name(product_name):
                     child.find('categoryPath')[len(child.find('categoryPath')) - 1][0].text is not None):
                       # Choose last element in categoryPath as the leaf categoryId
                       cat = child.find('categoryPath')[len(child.find('categoryPath')) - 1][0].text
+                      product_counter[cat]=product_counter.get(cat,0)+1
                       # Replace newline chars with spaces so fastText doesn't complain
                       name = child.find('name').text.replace('\n', ' ')
-                      output.write("__label__%s %s\n" % (cat, transform_name(name)))
-
+                      if product_counter[cat] > min_products:
+                        output.write("__label__%s %s\n" % (cat, transform_name(name)))
```

Comment:
I think this will skip the first `min_products` products in every category (so, if `min_products` if 5, and a particular category has 7 products in it, this will skip the first 5 and only write 2 products, instead of writing all 7).